### ISSUE TITLE
Incorrect comma insertion in dictionary definition when line wrapping

### ISSUE THREAD

**Issue by zacwest** [nicklockwood/SwiftFormat#885]
Given:

```swift
func doAThing() {
    thing.scan([String: [String: Observable<SomeCellModel>]]()) { (dict: [String: [String: Observable<SomeCellModel>]], update: Update) -> [String: [String: Observable<SomeCellModel>]] in

    }
}
```

Running this through `swiftformat --maxwidth 120` produces:

```swift
func doAThing() {
    thing
        .scan([String: [
            String: Observable<SomeCellModel>,
        ]](
        )) { (_: [String: [String: Observable<SomeCellModel>]], _: Update) -> [String: [String: Observable<SomeCellModel>]] in
        }
}
```

The dictionary parameter `[String: [String: Observable<ListCellModel>]]` has a comma inserted after the dictionary definition, causing a compilation error.

Notably, this does not occur if a trailing closure isn't used.

**Comment by nicklockwood** [comment]
@zacwest fixed in 0.47.13.

**Comment by zacwest** [comment]
I can confirm, thanks!
